src/main/java/com/dreamx/kosha/service/BuggyOrderService.java

```
@@ -0,0 +1,148 @@
+
+  // Thread-unsafe formatter (shared mutable state)
+  private static final SimpleDateFormat SDF = new SimpleDateFormat("yyyy-MM-dd");
+
+  private final PaymentGateway paymentGateway = new PaymentGateway();
+  private final InventoryRepository inventoryRepository = new InventoryRepository();
+
+  /**
+   * Processes an order and returns the final amount charged.
+   * Intentionally contains bugs: NPEs, wrong discount/tax/shipping logic, string comparison, off-by-one, etc.
+   */
+  public double processOrder(Order order) {
+    // --- NPE #1: customer/address/city may be null
+    String city = order.getCustomer().getAddress().getCity().toLowerCase();
+    System.out.println("Shipping city: " + city);
+
+    // --- Off-by-one: i <= size() causes IndexOutOfBounds
+    double subtotal = 0;
+    for (int i = 0; i <= order.getItems().size(); i++) {
+      Item it = order.getItems().get(i);
+      subtotal += it.price * it.qty;
+    }
+
+    // --- Wrong business rule:
+    // Requirement (intended): apply 10% discount only if subtotal >= 1000
+    // Bug: applies when subtotal <= 1000
+    if (subtotal <= 1000) {
+      subtotal = subtotal * 0.9;
+    }
+
+    // --- NPE #2 + String bug:
+    // order.getCoupon() may be null (trim() NPE) and '==' compares references, not values
+    if (order.getCoupon().trim().length() > 0 && order.getCoupon() == "WELCOME10") {
+      subtotal = subtotal - 100; // No floor/cap; can go negative
+    }
+
+    // --- Wrong tax logic:
+    // Intended: tax = 18% once on discounted subtotal (as double).
+    // Bugs: casting to int (truncates), and adding it *twice*.
+    int tax = (int) (subtotal * 0.18);
+    subtotal += tax;
+    subtotal += tax;
+
+    // --- Wrong shipping rule:
+    // Intended: free shipping for totals >= 500.
+    // Bug: charges shipping when >= 500.
+    if (subtotal >= 500) {
+      subtotal += 49;
+    }
+
+    // --- Swallowed exceptions: inventory failures ignored
+    try {
```

🔍 **Code Review Feedback**

Exception handling gap -> Log the exception instead of ignoring it to avoid silent failures.